### PR TITLE
[wip] change volume_status to volume_gb

### PR DESCRIPTION
**Pull request by alexeymyltsev**
https://github.com/openstack-exporter/openstack-exporter/issues/180

### CODE REVIEW

**Comment by niedbalski** [comment]
@alexeymyltsev thanks for this patch.

I think we'll need to create a deprecation cadence process for metrics that extends the metrics
for a X+1 release to avoid disruptive changes to land without further notice, which includes a few things:

1. A new README.md section named "deprecated", which has (metric name | last release | deprecation on | notes), so as an example
this metric will be last supported in 1.4 | 1.5 won't have it | deprecated in favor of volume_gb.
2. A --skip-deprecated flag that will avoid deprecated metrics, extend the default metric (deprecated = false) that will show a log.warn message saying this metric has been deprecated and will be removed as of release X.Y 
3. Extend the default metric to include a deprecated{version, bool} (nil, false) by default.

@alexeymyltsev ideas/opinions?

**Comment by alexeymyltsev** [comment]
@niedbalski your plan looks good for me. 
This process should make exports more mature.

Should we implement it in this PR or will be added new one?

**Comment by niedbalski** [comment]
@alexeymyltsev are you planning to work on it? Its fine if you want to include that in this PR. (no objections on my end)

**Comment by alexeymyltsev** [comment]
@niedbalski ok, I will work on it.

File: exporters/cinder.go
Comment (brolga) [inline exporters/cinder.go:27]: I think the label "size" ist not neede wens size is already in the metric.

Comment (alexeymyltsev) [reply exporters/cinder.go]: you are right. 
Below commit remove it.

**Comment by brolga** [comment]
The is another thing that comes into my mind. 
Please bear with me.

 Prometheus says on its webpage
```Metrics must use base units (e.g. seconds, bytes) and leave converting them to something more readable to graphing tools.```

**Comment by alexeymyltsev** [comment]
@brolga 
> Prometheus says on its webpage
Metrics must use base units (e.g. seconds, bytes) and leave converting them to something more readable to graphing tools.

you mean we should use bytes instead gb?
Unfortunately OpenStack for volumes size return value in GB and all another metrics already have suffix _gb.
I think we can skip this requirements from Prometheus.

**Comment by brolga** [comment]
This is ok for me.

**Comment by niedbalski** [comment]
Thanks! I am going to set this review as work in progress @alexeymyltsev , until we have the deprecated metrics support.

**Review by niedbalski — CHANGES_REQUESTED** [review]
On hold.

File: exporters/cinder.go
Comment (niedbalski) [inline exporters/cinder.go:126]: I think the correct way for this would be to move this metric into a separated function and simply let the if exporter.isDeprecatedMetric to check for this condition.

File: README.md
Comment (niedbalski) [inline README.md:116]: why the trailing whitespaces?

Comment (niedbalski) [inline README.md:116]: why the trailing whitespaces?

File: exporters/cinder.go
Comment (niedbalski) [inline exporters/cinder.go:61]: 1.4, 1.5 we will remove it :-)

**Review by niedbalski — CHANGES_REQUESTED** [review]
Hey @alexeymyltsev great work on this. I left a couple of comments to be addressed before merging.

**Comment by niedbalski** [comment]
@alexeymyltsev added a couple of comments. I'd like to release 1.4.0 with this patch included , so holding on this.

File: README.md
Comment (alexeymyltsev) [reply README.md:116]: I think view like this 
image could be better.
Should I remove this changes?

File: exporters/cinder.go
Comment (alexeymyltsev) [reply exporters/cinder.go:61]: Ok I will add in warning that metric will be removed in next release

**Comment by alexeymyltsev** [comment]
I have added separate function for volume_status.

**Comment by alexeymyltsev** [comment]
@niedbalski is there some additional comments for this PR?

**Review by niedbalski — APPROVED** [review]
LGTM

**Pull request merged**
